_posts/2022-04-13-webhttp.markdown

```diff
@@ -0,0 +1,148 @@
+
+We'll want a simple service. First the contract:
+
+```csharp
+[ServiceContract]
+public interface IWebApi
+{
+    [OperationContract]
+    [WebGet(UriTemplate = "/hello")]
+    string PathEcho();
+}
+```
+
+And then the implementation:
+
+```csharp
+public class WebApi : IWebApi
+{
+    public string PathEcho() => "Hello World!";
+}
+```
+
+This should look familiar to you if you've used these parts of WCF before. Swap out all of the startup logic in Program.cs with this:
+
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+
+builder.WebHost.ConfigureKestrel(options =>
+{
+    options.AllowSynchronousIO = true;
+    options.ListenLocalhost(8080);
+});
+
+builder.Services.AddServiceModelWebServices();
+
+var app = builder.Build();
+
+app.UseServiceModel(builder =>
+{
+    builder.AddService<WebApi>();
+    builder.AddServiceWebEndpoint<WebApi, IWebApi>("api");
+});
+
+app.Run();
+```
+
+This will look familiar to you if you've used ASP.NET Core before. The CoreWCF specific calls here are AddServiceModelWebServices and UseServiceModel. With that in place go ahead and run the app: `dotnet run`. Once it's running if you navigate to <http://localhost:8080/api/hello> in a browser you will see the response.
+
+#### Documentation
+
+The REST support in WCF would generate help pages about the API if the HelpEnabled setting was enabled on one or more bindings. These days most REST APIs are generally documented with OpenAPI, so in light of this the generated help page in CoreWCF is actually Swagger UI. As an aside the support for OpenAPI here is less robust than it would be in something like ASP.NET Core using the Swashbuckle, but should hopefully be a useful tool regardless. With that out of the way let's explore how to work with it:
+
+First we'll want to enable help for our contract. Change the AddServiceWebEndpoint call to be:
+
+```csharp
+builder.AddServiceWebEndpoint<WebApi, IWebApi>("api", behavior =>
+{
+    behavior.HelpEnabled = true;
+});
+```
+
+Note that this is also how you would configure the WebHttpBehavior for any other reason as well. We'll also need another Nuget package. Add it by running: `dotnet add package Swashbuckle.AspNetCore.SwaggerUI`. Back in the startup we'll need to add something else to services:
+
+```csharp
+builder.Services.AddSingleton(new SwaggerOptions());
+```
+
+And some middleware:
+
+```csharp
+app.UseMiddleware<SwaggerMiddleware>();
+app.UseSwaggerUI();
+```
+
+Now CoreWCF will try and intuit as much information as it can about your API for documentation, but sometimes you will have to feed it information directly. This will be done with attributes. Let's add the our first attribute to the service contract:
```

Comment:
>Now CoreWCF will try and intuit as much information as it can about your API for documentation, but sometimes you will have to feed it information directly. This will be  [](http://example.com/codeflow?start=0&length=170)

The last sentence on this line is a bit jumbled. Looks like you wrote one thing, then edited to reword it. "Let's add **the our** first attribute to the service contract"